pymeasure/instruments/keithley/keithley2306.py

```diff
@@ -0,0 +1,683 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2021 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import truncated_range, strict_discrete_set
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class Channel():
+    """ Implementation of a Keithley 2306 channel. """
+
+    enabled = Instrument.control(
+        ":OUTPUT<ch>:STAT?", ":OUTPUT<ch>:STAT %d",
+        """A boolean property that controls whether the output is enabled, takes
+        values True or False. """,
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True
+    )
+
+    bandwidth = Instrument.control(
+        ":OUTPUT<ch>:BAND?", ":OUTPUT<ch>:BAND %s",
+        """A string property that controls the output bandwidth when the output
+        is enabled and the current range is set to 5 A. Takes values 'HIGH' or
+        'LOW'. If the output is disabled or the current range is set to 5 mA the
+        bandwidth is 'LOW'. """,
+        validator=strict_discrete_set,
+        values={'low': 'LOW', 'high': 'HIGH'},
+        map_values=True,
+    )
+
+    sense_mode = Instrument.control(
+        ":SENS<ch>:FUNC?", ":SENS<ch>:FUNC \"%s\"",
```

Comment:
Just a Python detail you are maybe already aware of anyway, you can mix and match quotes in strings, so the second string here does not need escapes, `':SENS<ch>:FUNC "%s"'` should work, too.